contracts/contracts/utils/AerodromeAMOQuoter.sol

```diff
@@ -0,0 +1,316 @@
+                uint256 balanceWETH;
+                uint256 amountWETH;
+
+                // solhint-disable-next-line no-inline-assembly
+                assembly {
+                    balanceWETH := mload(add(reason, 0x24))
+                    amountWETH := mload(add(reason, 0x44))
+                }
+                return
+                    RebalanceStatus({
+                        reason: RevertReasons.NotEnoughWethForSwap,
+                        currentPoolWETHShare: 0,
+                        allowedWETHShareStart: 0,
+                        allowedWETHShareEnd: 0,
+                        currentTick: 0,
+                        balanceWETH: balanceWETH,
+                        amountWETH: amountWETH,
+                        revertMessage: ""
+                    });
+            }
+
+            // Case 4: Not enough WETH liquidity
+            if (
+                receivedSelector == IAMOStrategy.NotEnoughWethLiquidity.selector
+            ) {
+                return
+                    RebalanceStatus({
+                        reason: RevertReasons.NotEnoughWethLiquidity,
+                        currentPoolWETHShare: 0,
+                        allowedWETHShareStart: 0,
+                        allowedWETHShareEnd: 0,
+                        currentTick: 0,
+                        balanceWETH: 0,
+                        amountWETH: 0,
+                        revertMessage: ""
+                    });
+            }
+
+            // Case 5: Unexpected error
+            return
+                RebalanceStatus({
+                    reason: RevertReasons.UnexpectedError,
+                    currentPoolWETHShare: 0,
+                    allowedWETHShareStart: 0,
+                    allowedWETHShareEnd: 0,
+                    currentTick: 0,
+                    balanceWETH: 0,
+                    amountWETH: 0,
+                    revertMessage: abi.decode(reason, (string))
+                });
+        }
+    }
+}
+
+contract AerodromeAMOQuoter {
+    QuoterHelper public quoterHelper;
+
+    constructor() {
+        quoterHelper = new QuoterHelper();
+    }
+
+    struct Data {
+        uint256 amount;
+        uint256 iterations;
+    }
+
+    event ValueFound(uint256 value, uint256 iterations);
+    event ValueNotFound(string message);
+
+    function quoteAmountToSwapBeforeRebalance(
+        address strategy,
+        bool swapWETHForOETHB,
+        uint256 minAmount,
+        uint256 maxAmount,
+        uint256 maxIterations
+    ) public returns (Data memory data) {
+        QuoterHelper.QuoterParams memory params = QuoterHelper.QuoterParams({
+            strategy: strategy,
+            swapWETHForOETHB: swapWETHForOETHB,
+            minAmount: minAmount,
+            maxAmount: maxAmount,
+            maxIterations: maxIterations
+        });
+        try quoterHelper.getAmountToSwapBeforeRebalance(params) {
+            revert("Previous call should only revert, it cannot succeed");
```

Comment:
pretty cool that this always reverts. As to not change any state in the strategy / pool contracts 👍